frontend/src/components/nav/navbar.js

```diff
@@ -56,6 +57,15 @@ class NavBar extends React.Component {
         }
     }
 
+    debounceSearch(func, delay) {
+        let inDebounce;
+        return (e) => {
+            e.persist();
+            clearTimeout(inDebounce);
```

Comment:
You should actually check if that has been set. There should be two states for this variable: set or not set. I think it makes more sense to initialize the variable as null. Then you can do this:

if (timerId !== null) {
  window.clearTimeout(timerId);
}

This way you only clear the timer if it has been set. It's also better to do window.clearInterval instead of relying on the global scope.